Stripe/STPAnalyticsClient.m

```diff
@@ -0,0 +1,68 @@
+//
+//  STPAnalyticsClient.m
+//  Stripe
+//
+//  Created by Ben Guo on 4/22/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+#import "STPAnalyticsClient.h"
+#import "STPFormEncoder.h"
+#import "NSMutableURLRequest+Stripe.h"
+#import "STPAPIClient.h"
+
+@interface Stripe (Private_Analytics)
++ (BOOL)shouldCollectAnalytics;
+@end
+
+@interface STPAnalyticsClient()
+
+@property (nonatomic, readwrite) NSURLSession *urlSession;
+@property (nonatomic, readwrite) NSURL *baseURL;
+
+@end
+
+@implementation STPAnalyticsClient
+
++ (BOOL)shouldCollectAnalytics {
+    return NSClassFromString(@"XCTest") == nil && [Stripe shouldCollectAnalytics];
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+        _urlSession = [NSURLSession sessionWithConfiguration:config];
+        _baseURL = [NSURL URLWithString:@"https://q.stripe.com"];
+    }
+    return self;
+}
+
+- (void)logRUMWithTokenType:(NSString *)tokenType
+                   response:(NSURLResponse *)response
+                      start:(NSDate *)startTime
+                        end:(NSDate *)endTime {
+    if (![[self class] shouldCollectAnalytics]) {
+        return;
+    }
+    if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+        return;
+    }
+    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+    NSNumber *start = @((NSInteger)[startTime timeIntervalSince1970]);
```

Comment:
The browser-side version of this is reporting milliseconds. Any chance you could * 1000 before the integer cast?